test-case/verify-ucm-config.sh

```diff
@@ -0,0 +1,123 @@
+    dlogc "alsaucm -c '$cardname' set _verb $verb list _devices"
+    devices=$(alsaucm -c "$cardname" set _verb $verb list _devices)
+    [[ $? -ne 0 ]] && dloge "list device failed" && return 1
+    ((descline=0))
+    local OLD_IFS="$IFS"
+    IFS=$'\n'
+    for device in $devices; do
+        if [ $descline -eq "0" ]; then
+            device=`echo $device |awk -F ": " '{print $2}'`
+            [[ -z "$verb" ]] && dlogw "device is null, skipping" && continue
+            dlogi "found device: $device"
+
+	    # enable the device and check if the device is really enabled or not
+            dlogc "alsaucm -c '$cardname' set _verb '$verb' set _enadev '$device' list1 _enadevs"
+            ret=`alsaucm -c "$cardname" set _verb "$verb" set _enadev "$device" list1 _enadevs`
+            [[ $? -ne 0 ]] && dloge "enable '$verb':'$device' failed" && IFS="$OLD_IFS" && return 1
+            ret=`echo $ret |grep "$device" 2>/dev/null`
+            [[ -z "$ret" ]] && dloge "'$verb':'$device' is not in the enabled device list" && IFS="$OLD_IFS" && return 1
+
+	    # enable the device and setup the controls to check if the controls can be set correctly or not
+            dlogc "alsaucm -c '$cardname' set _verb '$verb' set _enadev '$device' reload"
+            alsaucm -c "$cardname" set _verb "$verb" set _enadev "$device" reload
+            [[ $? -ne 0 ]] && dloge "reload '$verb':'$device' setting failed!" && IFS="$OLD_IFS" && return 1
+
+            # device format: the first line is device and the second line is description
+            # Let's skip the description
+            ((descline=1))
+        else
+            ((descline=0))
+        fi
+    done
+    IFS="$OLD_IFS"
+    return 0
+}
+
+ucm_folder=/usr/share/alsa/ucm
+
+sofcard=${SOFCARD:-0}
+cardname=$(sof-dump-status.py -s $sofcard)
+[[ $? -ne 0 ]] && exit 1
+
+# TODO: check the audio card dmi name (dmi name is not recommanded).
+ucm_file=${ucm_folder}/${cardname}/${cardname}.conf
+
+[[ ! -e "${ucm_file}" ]] && dloge "ucm configuration: $ucm_file not exists!" && exit 2
+
+# currently we only supports old ucm configuration.
+# only after alsa-lib, alsa-utils and pulseaudio are updated, UCMv2 will be supported.
+# check ucm version
+[[ "$(grep "Syntax 2" "${ucm_file}")" ]] && dloge "not support UCMv2 currently" && exit 2
+
+# use alsaucm to check the ucm configurations
+dlogc "alsaucm open '$cardname'"
+# maybe use cardname let alsaucm choose ucm file
+alsaucm open "$cardname"
+[[ $? -ne 0 ]] && dloge "open card '$cardname' failed!" && exit 1
+
+dlogc "alsaucm -c "$cardname" reload"
+alsaucm -c "$cardname" reload
+[[ $? -ne 0 ]] && dloge "card '$cardname' reload failed. Please check the card amixer setting." && exit 1
+
+dlogc "alsaucm -c '$cardname' list _verbs"
+verbs=`alsaucm -c "$cardname" list _verbs`
+[[ $? -ne 0 ]] && dloge "'$cardname' list _verbs failed!" && exit 1
+[[ -z "$verbs" ]] && dloge "'$cardname' ucm has no verbs!" && exit 1
+((i=0))
+
+OLD_IFS="$IFS"
```

Comment:
I also see `OLD_IFS` with local prefix in `func_verify_verb_device()`. Could you check if you are miss use or use in purpose. 
Maybe change a name to avoid confusion.

Reply:
That one is local version. I will define this one as local too.